python/foglamp/services/core/api/notification.py

```diff
@@ -0,0 +1,517 @@
+        await config_mgr.create_child_category(name, [category_name])
+
+        # If deliveryConfig is in POST data, then update the value for each config item
+        if delivery_config is not None:
+            if not isinstance(delivery_config, dict):
+                raise ValueError('deliveryConfig must be a JSON object')
+            for k, v in delivery_config.items():
+                await config_mgr.set_category_item_value_entry("delivery{}".format(name), k, v['value'])
+    except Exception as ex:
+        await _delete_configuration(storage, config_mgr, name)  # Revert configuration entry
+        _logger.exception("Failed to create notification configuration. %s", str(ex))
+        raise web.HTTPInternalServerError(reason='Failed to create notification configuration.')
+
+
+async def _update_configurations(config_mgr, name, notification_config, rule_config, delivery_config):
+    try:
+        # ---------------- Update main notification
+        category_desc = notification_config['description']['default']
+        await config_mgr.create_category(category_name=name,
+                                         category_description=category_desc,
+                                         category_value=notification_config,
+                                         keep_original_items=True)
+
+        # ---------------- Replace rule configuration
+        if rule_config != {}:
+            category_desc = rule_config['plugin']['description']
+            category_name = "rule{}".format(name)
+            await config_mgr._update_category(category_name=category_name,
+                                             category_description=category_desc,
+                                             category_val=rule_config)
+
+        # ---------------- Replace delivery configuration
+        if delivery_config != {}:
+            category_desc = delivery_config['plugin']['description']
+            category_name = "delivery{}".format(name)
+            await config_mgr._update_category(category_name=category_name,
+                                             category_description=category_desc,
+                                             category_val=delivery_config)
+    except Exception as ex:
+        _logger.exception("Failed to update notification configuration. %s", str(ex))
+        raise web.HTTPInternalServerError(reason='Failed to update notification configuration.')
+
+
+async def _delete_configuration(storage, config_mgr, name):
+    current_config = await config_mgr._read_category_val(name)
+    if not current_config:
+        return
+    rule = current_config['rule']['value']
+    channel = current_config['channel']['value']
+
+    await _delete_configuration_category(storage, name)
+    await _delete_configuration_category(storage, "rule{}".format(name))
+    await _delete_configuration_category(storage, "delivery{}".format(name))
+    await _delete_parent_child_configuration(storage, "Notifications", name)
```

Comment:
For delete parent child relationshjp you can also use `await config_mgr.delete_child_category(category_name, child_category)`